src/openfermion/measurements/_probability_distributions.py

```diff
@@ -0,0 +1,363 @@
+#   Licensed under the Apache License, Version 2.0 (the "License");
+#   you may not use this file except in compliance with the License.
+#   You may obtain a copy of the License at
+#
+#       http://www.apache.org/licenses/LICENSE-2.0
+#
+#   Unless required by applicable law or agreed to in writing, software
+#   distributed under the License is distributed on an "AS IS" BASIS,
+#   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#   See the License for the specific language governing permissions and
+#   limitations under the License.
+"""
+Classes to store probability distributions for estimation
+of random variables.
+"""
+import numpy
+from scipy import sparse
+from numpy import cos, sin, pi
+from openfermion.config import EQ_TOLERANCE
+
+
+class FourierProbabilityDist(object):
+    """
+    Stores a multivariant Fourier representation of a periodic function:
+
+    f(phi_0,phi_1,...) = sum_j A_j f_j
```

Comment:
I'm sorry, @obriente , but I still do not understand. You're claiming here that this class represents a _function_. Here, you wrote the function as
```
f(phi_0,phi_1,...) = sum_j A_j f_j
```
so `f` is the function being represented by the class.

In the docstring of `__init__`, you refer to the "distribution" of the `A_j`. But for the function `f` defined above, there's no such thing as a "distribution" for the `A_j`; there is just the `A_j`.

So is this class representing the function
```
f(phi_0,phi_1,...) = sum_j A_j f_j
```
or is it representing something else?

Reply:
Oh, this is a very good point, the docstring is flat wrong. This class should store a representation of a distribution P(A_0,A_1,...,\phi_0,\phi_1,...), and in particular in this class we approximate P=P(A_0,A_1,...)\prod_jP(\phi_j), and store data about P(A_0,A_1,...) and P(phi_j).

Thanks for pointing this out, I feel like this might have been the source of a lot of confusion :(. I'll rewrite the docstring now.